### ISSUE TITLE
ObjectStore.list() should not raise NotFoundError if no objects found

### ISSUE THREAD

**Issue by skewty** [nats-io/nats.py#586]
### Observed behavior

`NotFoundError` is raised. This exception is unexpected (not mentioned in docstring) and non-pythonic.

### Expected behavior

Empty list is returned. This meets the contract of the docstring and method signature and is pythonic.

### Server and client version

nats-py 2.8.0

nats-server 2.10.17

**Comment by skewty** [comment]
@caspervonb the current behavior seems intentional. Would adding a `raise_on_empty: bool = True` kwarg be an acceptable "fix" as changing this would be a breaking change.  Should this be deferred to a v3 release?

**Comment by caspervonb** [comment]
Thanks for the report — same shape as the legacy \`KeyValue.keys()\` issue in #582.

Changing \`ObjectStore.list()\` to return \`[]\` on empty is a breaking API change for callers that catch \`NotFoundError\` as the empty-store signal, so we're going to leave the legacy \`nats.js.ObjectStore\` behavior as-is.

A dedicated \`nats-object\` package is planned and will mirror the design pattern of the upcoming \`nats-key-value\` package (#844) — an iterator-style API that returns empty naturally instead of raising. That'll be the migration path for code that finds the current raise-on-empty awkward. It isn't released yet, but once it lands the recommended migration will be straightforward.

Closing as won't fix for the legacy package.
